cranelift-codegen/meta-python/isa/x86/encodings.py

```diff
@@ -198,9 +198,18 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
 # TODO: band_imm.i64 with an unsigned 32-bit immediate can be encoded as
 # band_imm.i32. Can even use the single-byte immediate for 0xffff_ffXX masks.
 
+# 32-bit rules for iconst variants with immediate 0.
+X86_32.enc(base.iconst.i32, *r.u_id_z(0x31))
+
 # Immediate constants.
 X86_32.enc(base.iconst.i32, *r.pu_id(0xb8))
 
+# 64-bit rules for iconst variants with immediate 0.
+X86_64.enc(base.iconst.i32, *r.u_id_z.rex(0x31))
```

Comment:
I have these added before the more broad recipes so they get picked first, because (if I understood correctly), the squeezing step of binemit isn't run for filetests. Since this seems like an extra step if optimizations are enabled, should this be moved later so a more permissive catchall could apply first?

Reply:
ended up putting these rules later so first-pass encoding can happen faster, then opting a few tests into `opt_level=best` to test that squeezing picks these smaller encodings. "instruction length" isn't a great selection metric for these, because they're better alternatives for other reasons, but it'll do